src/index.ts

```diff
@@ -0,0 +1,211 @@
+
+export const beautifier: Beautifier = {
+  name: "clang-format",
+  package: pkg,
+  badges: [
+    {
+      description: "Build Status",
+      url:
+        "https://travis-ci.com/Unibeautify/beautifier-clang-format.svg?branch=master",
+      href: "https://travis-ci.com/Unibeautify/beautifier-clang-format",
+    },
+    {
+      description: "Test Coverage",
+      url:
+        "https://api.codeclimate.com/v1/badges/8b4e5b19de96f4744ffa/test_coverage",
+      href:
+        "https://codeclimate.com/github/Unibeautify/beautifier-clang-format/test_covera" +
+        "ge",
+    },
+    {
+      description: "Maintainability",
+      url:
+        "https://api.codeclimate.com/v1/badges/8b4e5b19de96f4744ffa/maintainability",
+      href:
+        "https://codeclimate.com/github/Unibeautify/beautifier-clang-format/maintainabi" +
+        "lity",
+    },
+    {
+      description: "Greenkeeper",
+      url:
+        "https://badges.greenkeeper.io/Unibeautify/beautifier-clang-format.svg",
+      href: "https://greenkeeper.io/",
+    },
+  ],
+  options: {
+    "C": true,
+    "C++": true,
+    Java: true,
+    "Objective-C": true
+  },
+  dependencies: [
+    {
+      type: DependencyType.Executable,
+      name: "clang-format",
+      program: "clang-format",
+      parseVersion: [/clang-format version (\d+\.\d+\.\d+)/],
+      homepageUrl: "https://clang.llvm.org/docs/ClangFormat.html",
+      installationUrl: "https://clang.llvm.org/docs/ClangFormat.html",
+      bugsUrl: "https://clang.llvm.org/docs/ClangFormat.html",
+      badges: [
+        {
+          description: "",
+          url: "",
+          href: "",
+        },
+      ],
+    },
+  ],
+  resolveConfig: ({ filePath, projectPath }) => {
+    const configFiles: string[] = [".clang-format", "_clang-format"];
+    return findFile({
+      finishPath: projectPath,
+      startPath: filePath,
+      fileNames: configFiles,
+    })
+      .then(configFile => ({ filePath: configFile }))
+      .catch(err => {
+        // tslint:disable-next-line no-console
+        console.log(err);
+        return Promise.resolve({});
+      });
+  },
+  beautify({
+    text,
+    dependencies,
+    filePath,
+    beautifierConfig,
+  }: BeautifierBeautifyData) {
+    const clangFormat = dependencies.get<ExecutableDependency>("clang-format");
+    const basePath: string = os.tmpdir();
+    let fileExtension;
+    if (filePath) {
+      fileExtension = path.extname(filePath);
+    }
+    const config =
+      beautifierConfig && beautifierConfig.filePath
+        ? `--style=${beautifierConfig.filePath}`
+        : "";
+    // tslint:disable-next-line no-console
+    console.log(`Using config: ${config}`);
+    return tmpFile({ postfix: fileExtension }).then(filePath =>
+      writeFile(filePath, text).then(() =>
+        clangFormat
+          .run({
+            args: relativizePaths(
+              ["-style=file", `-assume-filename=${filePath}`, "-i", filePath],
```

Comment:
Background reading notes for me:
>   -assume-filename=<string> - When reading from stdin, clang-format assumes this
                              filename to look for a style config file (with
                              -style=file) and to determine the language.

Reply:
From https://clang.llvm.org/docs/ClangFormat.html

> If no arguments are specified, it formats the code from standard input
and writes the result to the standard output.
If <file>s are given, it reformats the files. If -i is specified
together with <file>s, the files are edited in-place. Otherwise, the
result is written to the standard output.

- [ ] Could we remove `-i` argument and let it use the standard output?

Reply:
Do we need the `-assume-filename` as `filePath` is already used at the end for the `<file>` CLI argument?